pkg/factsengine/entities/fact_value.go

```diff
@@ -44,20 +57,34 @@ func NewFactValue(factInterface interface{}) (FactValue, error) {
 	case map[string]interface{}:
 		newMap := make(map[string]FactValue)
 		for key, mapValue := range value {
-			newValue, err := NewFactValue(mapValue)
+			newValue, err := NewFactValueWithConf(mapValue, conf)
 			if err != nil {
 				return nil, err
 			}
+			if conf.SnakeCaseKeys {
+				key = strcase.ToSnake(key)
+			}
 			newMap[key] = newValue
 		}
 		return &FactValueMap{Value: newMap}, nil
-	case bool, int, int32, int64, uint, uint32, uint64, float32, float64, string:
+	case bool, int, int32, int64, uint, uint32, uint64, float32, float64:
```

Comment:
PD: I tried to split the case entries for the current `int` and `float` adding more `case` clauses, and avoid using the `ParseStringToFactValue` function for them, but it slightly changes the behaviour. For example, now, float numbers without decimals are converted to `int`, but if we get `float64` in the case, we would convert it to `float`.

This is how it would look like:
```
	case bool:
		return &FactValueBool{Value: value}, nil
	case int:
		return &FactValueInt{Value: value}, nil
	case int32:
		return &FactValueInt{Value: int(value)}, nil
	case int64:
		return &FactValueInt{Value: int(value)}, nil
	case uint:
		return &FactValueInt{Value: int(value)}, nil
	case uint32:
		return &FactValueInt{Value: int(value)}, nil
	case uint64:
		return &FactValueInt{Value: int(value)}, nil
	case float32:
		return &FactValueFloat{Value: float64(value)}, nil
	case float64:
		return &FactValueFloat{Value: value}, nil
	case string:
		if conf.StringConversion {
			return ParseStringToFactValue(value), nil
		}
		return &FactValueString{Value: value}, nil
```
But we might have some issues converting `uint64` to `int` in some corner case.
Wdyt?

Reply:
Does this behaviour change affect any of the existing gatherers?

Reply:
Yes, values like `30s` (which we have in cibadmin gatherer) pass to be of float type instead of int type

Reply:
I'm letting this as it is by now, to avoid any (even really not harmful) breaking change